src/main/java/io/frictionlessdata/datapackage/JSONBase.java

```diff
@@ -67,13 +71,13 @@
     private String hash = null;
 
     Dialect dialect;
-    private JSONArray sources = null;
-    private JSONArray licenses = null;
+    private ArrayNode sources = null;
+    private ArrayNode licenses = null;
```

Comment:
The appropriate classes to store sources and licenses should be introduced, such that this can be replaced with `List<Source> sources = null` and similarly for licenses.